python/tank/util/user_settings.py

```diff
@@ -45,48 +45,83 @@ def _init_singleton(self):
+            else:
+                logger.debug("Shotgun proxy (from settings): <empty>")
 
-        proxy = self._get_filtered_proxy(self.app_store_proxy)
-        if self.is_app_store_proxy_set():
-            logger.debug("App Store proxy: %s" % (proxy or "<empty>",))
+        # Retrieve the settings app store proxy string (possibly an empty string),
+        # or None when it is not present in the settings file.
+        self._app_store_proxy = self._get_value("app_store_http_proxy")
+
+        if self._app_store_proxy is None:
+            # Since the app store proxy is not present in the settings file,
+            # fall back on the operating system http proxy.
+            if self._system_proxy:
+                logger.debug("App Store proxy (from system): %s" % self._get_filtered_proxy(self._system_proxy))
+            else:
+                logger.debug("App Store proxy: <missing>")
         else:
-            logger.debug("App Store proxy: <missing>")
+            if self._app_store_proxy:
+                logger.debug("App Store proxy (from settings): %s" % self._get_filtered_proxy(self._app_store_proxy))
+            else:
+                logger.debug("App Store proxy (from settings): <empty>")
+
+    def is_shotgun_proxy_set(self):
+        """
+        :returns: ``True`` if ``http_proxy`` is present in the settings file,
+                  ``False`` otherwise.
+        """
+        return self._settings_proxy is not None
 
     @property
     def shotgun_proxy(self):
         """
-        :returns: The default proxy.
+        :returns: The settings http proxy when it is present and non-empty,
+                  or ``None`` when it is present and empty; otherwise,
+                  the operating system http proxy, or ``None`` when it is not defined.
+                  ``is_shotgun_proxy_set()`` can be used to disambiguate a ``None``.
         """
-
-        # Return the configuration settings http proxy string when it is specified;
-        # otherwise, return the operating system http proxy string.
-        return self._settings_proxy or self._system_proxy
+        if self._settings_proxy is None:
+            return self._system_proxy
+        else:
+            return self._settings_proxy or None
 
     def is_app_store_proxy_set(self):
         """
-        :returns: ``True`` if ``app_store_http_proxy`` is set, ``False`` otherwise.
+        :returns: ``True`` if ``app_store_http_proxy`` is present in the settings file,
+                  ``False`` otherwise.
         """
-        return self._is_setting_found("app_store_http_proxy")
+        return self._app_store_proxy is not None
 
     @property
     def app_store_proxy(self):
         """
-        :returns: The app store specific proxy. If ``None``, it means the setting is absent from the
-            file or set to an empty value. Use `is_app_store_proxy_set` to disambiguate.
+        :returns: The settings app store proxy when it is present and non-empty,
```

Comment:
similar to above wrt wording. Also noticed that these properties have `:returns:` - generally, we do properties without a returns so i suggest dropping that.